source/dub/semver.d

```diff
@@ -26,7 +26,7 @@ import std.conv;
 	Validates a version string according to the SemVer specification.
 */
 bool isValidVersion(string ver)
-{
+pure @nogc {
```

Comment:
Could you add it to `Version.toString` as well ? And maybe other methods, but this is the one I had to check in the previous commit as many `toString` are allocating.

Reply:
Yeah, that should definitely be done in more places. I'd do that in a separate pass through the code base, though. I did it here mainly because I was getting bogus stack traces involving the GC in the profiler.